src/snowflake/cli/_plugins/nativeapp/entities/application_package.py

```diff
@@ -546,8 +565,73 @@ def _bundle(self):
         )
         compiler = NativeAppCompiler(bundle_context)
         compiler.compile_artifacts()
+
+        # TODO should this merged into NativeAppCompiler?
+        self._inject_spcs(spcs_services)
+
         return bundle_map
 
+    def _inject_spcs(self, spcs_services: list[ServiceEntity]):
```

Comment:
Looks okay as a PoC, but we probably need to be more careful when we auto-modify the manifest and setup script